hal/inc/hal_dynalib_ble.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2018 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef HAL_DYNALIB_BLE_H
+#define	HAL_DYNALIB_BLE_H
+
+#include "dynalib.h"
+#include "hal_platform.h"
+
+#if HAL_PLATFORM_BLE
+
+#ifdef DYNALIB_EXPORT
+#include "ble_hal.h"
+#endif
+
+DYNALIB_BEGIN(hal_ble)
+
+DYNALIB_FN( 0,  hal_ble, ble_stack_init, int(void*) )
+DYNALIB_FN( 1,  hal_ble, ble_select_antenna, int(hal_ble_ant_type_t) )
+DYNALIB_FN( 2,  hal_ble, ble_set_callback_on_events, int(on_ble_evt_cb_t, void*) )
+DYNALIB_FN( 3,  hal_ble, ble_gap_set_device_address, int(const hal_ble_addr_t*) )
+DYNALIB_FN( 4,  hal_ble, ble_gap_get_device_address, int(hal_ble_addr_t*) )
+DYNALIB_FN( 5,  hal_ble, ble_gap_set_device_name, int(const uint8_t*, uint16_t) )
+DYNALIB_FN( 6,  hal_ble, ble_gap_get_device_name, int(uint8_t*, uint16_t*) )
+DYNALIB_FN( 7,  hal_ble, ble_gap_set_appearance, int(uint16_t) )
+DYNALIB_FN( 8,  hal_ble, ble_gap_get_appearance, int(uint16_t*) )
+DYNALIB_FN( 9, hal_ble, ble_gap_set_ppcp, int(const hal_ble_conn_params_t*, void*) )
+DYNALIB_FN( 10, hal_ble, ble_gap_get_ppcp, int(hal_ble_conn_params_t*, void*) )
+DYNALIB_FN( 11, hal_ble, ble_gap_add_whitelist, int(const hal_ble_addr_t*, uint8_t, void*) )
+DYNALIB_FN( 12, hal_ble, ble_gap_delete_whitelist, int(void*) )
+#define BROADCASTER_BASE_IDX 13
```

Comment:
Conditional base indices in dynalib tables have proven to be dangerous and difficult to maintain across platforms. Let's use continuous numbering here